error in speedcurve deployment recording blocks update of search index in stage/prod builds

When running the stage or production builds, we record the deployment in our Speedcurve account, but since that's done prior to updating the search index, if an error occurs, like the following:
```
Pinging Speedcurve Deploy API for 85742
Speedcurve Deploy note='59de7b32bb521ea6735b57130e532ffe3f9f3f56', detail='run_id=516115983 action=run7'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/runner/work/yari/yari/deployer/src/deployer/main.py", line 187, in speedcurve_deploy
    speedcurve_deploy_ping(
  File "/home/runner/work/yari/yari/deployer/src/deployer/speedcurve.py", line 37, in deploy_ping
    response.raise_for_status()
  File "/home/runner/.cache/pypoetry/virtualenvs/deployer-nuEoiHvI-py3.8/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.speedcurve.com/v1/deploys
Error: Process completed with exit code 1.
```
the search index is never updated and becomes out of sync with the documents that we've successfully deployed.

--- a/.github/workflows/prod-build.yml
+++ b/.github/workflows/prod-build.yml
@@ -215,6 +215,7 @@ jobs:
           DEPLOYER_ELASTICSEARCH_URL: ${{ secrets.DEPLOYER_PROD_ELASTICSEARCH_URL }}
 
         run: |
+
           if [ ${{ env.BUILD_ARCHIVED_CONTENT }} == "true" ]; then
             echo "Will build mdn/archived-content too"
             export CONTENT_ARCHIVED_ROOT=${{ github.workspace }}/mdn/archived-content/files
@@ -242,9 +243,6 @@ jobs:
 
           poetry run deployer upload ../client/build
           poetry run deployer update-lambda-functions ./aws-lambda
-          poetry run deployer speedcurve-deploy \
-            --note "${GITHUB_SHA}" \
-            --detail "run_id=${GITHUB_RUN_ID} action=${GITHUB_ACTION}"
 
           # TODO: Depending on how long the upload takes, consider switching to
           # use `--update` in this command. It will skip the index re-creation
@@ -253,3 +251,10 @@ jobs:
           # Consider doing something like a clean indexing on Sundays and
           # use `--update` on all the other week days.
           poetry run deployer search-index ../client/build --priority-prefix=en-us/docs/web
+
+          # Record the deployment in our Speedcurve account. This should always
+          # be done after the upload of the documents and Lambda functions, as well
+          # as the search-index update, so those aren't affected if this fails.
+          poetry run deployer speedcurve-deploy \
+            --note "${GITHUB_SHA}" \
+            --detail "run_id=${GITHUB_RUN_ID} action=${GITHUB_ACTION}"

--- a/.github/workflows/stage-build.yml
+++ b/.github/workflows/stage-build.yml
@@ -248,9 +248,6 @@ jobs:
 
           poetry run deployer upload ../client/build
           poetry run deployer update-lambda-functions ./aws-lambda
-          poetry run deployer speedcurve-deploy \
-            --note "${GITHUB_SHA}" \
-            --detail "run_id=${GITHUB_RUN_ID} action=${GITHUB_ACTION}"
 
           # TODO: Depending on how long the upload takes, consider switching to
           # use `--update` in this command. It will skip the index re-creation
@@ -259,3 +256,10 @@ jobs:
           # Consider doing something like a clean indexing on Sundays and
           # use `--update` on all the other week days.
           poetry run deployer search-index ../client/build --priority-prefix=en-us/docs/web
+
+          # Record the deployment in our Speedcurve account. This should always
+          # be done after the upload of the documents and Lambda functions, as well
+          # as the search-index update, so those aren't affected if this fails.
+          poetry run deployer speedcurve-deploy \
+            --note "${GITHUB_SHA}" \
+            --detail "run_id=${GITHUB_RUN_ID} action=${GITHUB_ACTION}"

